core/org.wso2.carbon.bootstrap/src/test/java/org/wso2/carbon/bootstrap/BootstrapTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017 WSO2, Inc. (http://wso2.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.bootstrap;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests functionality of Bootstrap class
+ */
+public class BootstrapTest {
+    private Bootstrap bootstrap = new Bootstrap();
+    private final Set<URL> classpath = new LinkedHashSet<URL>();
+    private File libFile;
+
+    /**
+     * Create necessary temporary files.
+     *
+     * @throws IOException for file operation failures
+     */
+    @BeforeMethod
+    public void setUp() throws IOException {
+        libFile = Files.createTempDirectory("libFile").toFile();
+    }
+
+    /**
+     * Test if loadClass throws ClassNotFoundException when trying to load
+     * a class that does not exist within the scope to load the rest of the app
+     *
+     * @throws Exception for failure of file operations or classloading
+     */
+    @Test(expectedExceptions = ClassNotFoundException.class)
+    public void testLoadClassException() throws Exception {
+        bootstrap.addFileUrl(libFile);
+        bootstrap.addClassPathEntries();
+        ClassLoader cl = new URLClassLoader(classpath.toArray(new URL[classpath.size()]));
```

Comment:
This classpath.toArray() always returns an empty array. Do we need to use the classpath variable in here?

Reply:
Fixed in 527e06d